cmd/kube-scheduler/app/server.go

```diff
@@ -189,10 +189,11 @@ func Run(cc schedulerserverconfig.CompletedConfig, stopCh <-chan struct{}) error
 
 	// Prepare the event broadcaster.
 	if cc.Broadcaster != nil && cc.EventClient != nil {
-		cc.Broadcaster.StartLogging(klog.V(6).Infof)
-		cc.Broadcaster.StartRecordingToSink(&v1core.EventSinkImpl{Interface: cc.EventClient.Events("")})
+		cc.Broadcaster.StartRecordingToSink(stopCh)
+	}
+	if cc.LeaderElectionBroadcaster != nil && cc.CoreEventClient != nil {
```

Comment:
this is needed to not migrate the whole leaderElection code (which in turn would result in migrating controller-manager)